### PR TITLE
Simplify `len()` comparisons

### PR DESCRIPTION
**Pull request by dosisod**
This PR removes certain `len()` comparisons when it would be faster/more succinct to do a truthy check instead, such as `len(x) == 0` vs `not x`.

This fixes the [`FURB115`](https://github.com/dosisod/refurb/blob/master/docs/checks.md#furb115-no-len-compare) errors in Refurb (see #14887).

### CODE REVIEW

File: mypy/checker.py [hunk 1]
```diff
@@ -5922,7 +5920,7 @@ def refine_away_none_in_comparison(
                 non_optional_types.append(typ)
 
         # Make sure we have a mixture of optional and non-optional types.
-        if len(non_optional_types) == 0 or len(non_optional_types) == len(chain_indices):
+        if not non_optional_types or len(non_optional_types) == len(chain_indices):
```
Comment (AlexWaygood) [inline mypy/checker.py:5923]: This one I'd prefer to keep as is, as it retains symmetry with the condition on the right-hand side of the `or`

File: mypy/checker.py [hunk 2]
```diff
@@ -6929,7 +6927,7 @@ def reduce_conditional_maps(type_maps: list[tuple[TypeMap, TypeMap]]) -> tuple[T
     We only retain the shared expression in the 'else' case because we don't actually know
     whether x was refined or y was refined -- only just that one of the two was refined.
     """
-    if len(type_maps) == 0:
+    if not type_maps:
```
Comment (AlexWaygood) [inline mypy/checker.py:6930]: This one I'd also prefer to keep as-is, as it retains symmetry with the `elif` condition immediately below, which has to use `len`

File: mypy/messages.py [hunk 3]
```diff
@@ -2944,7 +2944,7 @@ def make_inferred_type_note(
 def format_key_list(keys: list[str], *, short: bool = False) -> str:
     formatted_keys = [f'"{key}"' for key in keys]
     td = "" if short else "TypedDict "
-    if len(keys) == 0:
+    if not keys:
```
Comment (AlexWaygood) [inline mypy/messages.py:2947]: would prefer to keep this one as-is, as it retains symmetry with the `elif` immediately below

File: mypy/semanal.py [hunk 4]
```diff
@@ -2262,7 +2262,7 @@ def infer_metaclass_and_bases_from_compat_helpers(self, defn: ClassDef) -> None:
                     break
 
         metas = {defn.metaclass, with_meta_expr, add_meta_expr} - {None}
-        if len(metas) == 0:
+        if not metas:
```
Comment (AlexWaygood) [inline mypy/semanal.py:2265]: Would prefer to keep this one as is, as it retains symmetry with the `if len(metas) > 1` condition immediately below

File: mypy/semanal.py [hunk 5]
```diff
@@ -4602,7 +4602,7 @@ def visit_with_stmt(self, s: WithStmt) -> None:
         if s.unanalyzed_type:
             assert isinstance(s.unanalyzed_type, ProperType)
             actual_targets = [t for t in s.target if t is not None]
-            if len(actual_targets) == 0:
+            if not actual_targets:
```
Comment (AlexWaygood) [inline mypy/semanal.py:4605]: would prefer to keep this one as-is, as it retains symmetry with the `elif len(actual_targets) == 1` condition immediately below

File: mypy/semanal.py [hunk 6]
```diff
@@ -6707,7 +6707,7 @@ def halt(self, reason: str = ...) -> NoReturn:
     if body and isinstance(body[0], ExpressionStmt) and isinstance(body[0].expr, StrExpr):
         body = block.body[1:]
 
-    if len(body) == 0:
+    if not body:
```
Comment (AlexWaygood) [inline mypy/semanal.py:6710]: elif immediately below

**Review by AlexWaygood — APPROVED** [review]
LGTM

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**